inject.js

```diff
@@ -56,6 +56,18 @@ chrome.extension.sendMessage({}, function(response) {
       }
       this.initializeControls();
 
+      //Save the speed of current tab's video before leaving the page
+      //(ie. reloading page or going Backward/Forward in history)
+      //so that it doesn't restore the speed set by a video in another tab
+      //but instead, keeps the current tab's video speed the same,
+      //until changed by user or browser quits.
+      window.onbeforeunload = function(e) {
```

Comment:
Hmm, this one is tricky.. this behavior you're proposing makes sense for reload and back/forward, but not for new tabs, where the behavior would now be based on tab visibility order (which seems even worse) instead of actually making speed changes.